classes/class3/README.md

```diff
@@ -1 +1,364 @@
+```javascript
+app.engine('handlebars', exphbs({defaultLayout: 'main'}));
+app.set('view engine', 'handlebars');
+```
+Note that you're setting the default layout for the application to the `main.handlebars` file you just created. You can change the layout for a single template at render time with a simple option.
+
+Lastly, within the `index.js` routing file, replace:
+```javascript
+res.send("Welcome home!");
+```
+with
+```javascript
+res.render('home');
+```
+Restart your server, visit http://localhost:3000 and behold your template!
+
+At this point you are basically just writing plain HTML, let's fix that!
+Start by adding the following to your `home.handlebars` template:
+```html
+<ul>
+{{#each classes}}
+  <li>{{this}}</li>
+{{/each}}
+</ul>
+```
+Then change the contents of your `index.js` route again. This time to this:
+```javascript
+res.render("home", {"classes": [
+  "Olin.js",
+  "other class 1",
+  "other class 2",
+  "other class 3"]
+});
+```
+The second parameter that we are passing to the `render` function is the *context* for the template. That means that the word `classes` in the handlebars file is looking for an array named `classes` and it will create a new list item for *each* of the elements in the array. There's plenty more power in templates and we'll see them shortly.
+
+## MongoDB
+
+Node has the ability to store some information as variables. But all of the variables are stored in the process memory. If you restart the server (or the server crashes) the data will be wiped away. If you are storing information from users or information based on user input, we need a more permanent solution. That's very persistent storage comes in. A file, like `data.txt`, is a form of persistent storage. It can be written to, read from, and maintains information regardless of whether your server is running or has long since stopped. But files aren't useful for much more than storing log information because of their organization. When we want to retrieve data from persistent storage quickly, we need a solution like a database!
+
+MongoDB is a NoSQL database. It stores data in JSON-like documents, is relatively performant and offers useful features for production level systems. We're going to use it for these reasons, but also because it is very easy to integrate with Node and doesn't require learning too much about data storage. Alternatives, for example the ubiquitous MySQL, are also very common database solutions, but tend to take a bit more time to get set up in a Node environment and do not have as easy abstraction layers to shield you from interacting with the database directly (through the wonders of SQL). You will likely see them in any internships or jobs doing web-dev, but MongoDB more than suits our purposes and will teach you the basics of data storage without taking a couple weeks of class time to get comfortable with. So, let's get Mongo running and see what we can do with it:
```

Comment:
NoSQL database. It -> NoSQL database, which means that instead of tabular data, it

Might put MySQL and PostgreSQL both as examples.

"and do not have as easy abstraction layers to shield you from interacting with the database directly (through the wonders of SQL)" Not sure what you're trying to say here.